scheduler/system_sched.go

```diff
@@ -127,6 +129,19 @@ func (s *SystemScheduler) process() (bool, error) {
 		s.logger.Printf("[DEBUG] sched: %#v: rolling update limit reached, next eval '%s' created", s.eval, s.nextEval.ID)
 	}
 
+	// If there are failed allocations, we need to create a blocked evaluation
+	// to place the failed allocations when resources become available.
+	if len(s.plan.FailedAllocs) != 0 && s.blocked == nil {
+		e := s.ctx.Eligibility()
+		classes := e.GetClasses()
+		s.blocked = s.eval.BlockedEval(classes, e.HasEscaped())
```

Comment:
Does the system scheduler ever make sense to be blocked? If it fails constraints it just doesn't place on that node. e.g. if I say `ssd=true` then its not blocked, that node is just never eligible for the system job.